src/helpers/client/safari-client.ts

```diff
@@ -124,13 +142,34 @@ export class SafariClient implements Client {
 
   /**
    * Listens for 'AsperaDesktop.Response' events from the document,
-   * and delegates handling to the handleResponseEvent method.
+   * and delegates handling to the handleResponse method.
    */
   private listenResponseEvents() {
     document.addEventListener('AsperaDesktop.Response', (event: CustomEvent<JSONRPCResponse>) => {
-      this.handleResponseEvent(event.detail);
+      this.handleResponse(event.detail);
     });
   }
+
+  /**
+   * Listens for 'AsperaDesktop.TransferActivity' events from the document,
+   * and delegates handling to the handleTransferActivity method.
+   */
+  private listenTransferActivityEvents() {
+    document.addEventListener('AsperaDesktop.TransferActivity', (event: any) => {
+      asperaDesktop.activityTracking.handleTransferActivity(event.detail);
+    });
+  }
+
+  /**
+   * Sends a keep alive ping every 3 seconds.
+   */
+  private keepAlive() {
+    this.dispatchEvent(SafariExtensionEventType.Ping);
+
+    setTimeout(() => {
```

Comment:
Is there any risk of the constructor (SafariClient) being called multiple times and accidentally creating multiple dispatchEvents?  If so should this timeout be placed on the global class and cleaned up in the constructor (like if set clearTimeout and start a new one?

Reply:
In a normal scenario the constructor is called once when initializing the singleton (line 175)

Reply:
If no risk of leaks then this is fine. But if possible we should probably put a global instance and avoid this callback loop getting leaked onto multiple threads